### DISCUSSION TITLE
version for mac

### DISCUSSION THREAD

**Discussion by ochampi** [beeware/briefcase#1962] [Q&A]
I install briefcase on my mac  with visual code . When I run briefcase the app run up but the window is blank nether menu or any widget show up. 
Probably a version issue but I don't know wich on used.  
mac os sonoma 14.6.1   
python 3.12.2

**Comment by freakboy3742** [comment]
It's impossible to say what has happened based on the detail you've provided. My best guess is that either:
1. You haven't saved your changes to app.py
2. If you're using `briefcase run`, you haven't used the `-u` option to update your app's code.

It's highly unlikely that this is version related, or related to your choice of editor. 

My suggestion is that you start with the [Beeware Tutorial](https://docs.beeware.org), and follow those instructions *precisely*. If you have a problem, then you can tell us exactly where in the process you're seeing a problem.

**Comment by ochampi** [comment]
hi thanks for your answer. So I followed the beeware tutorial as it is. I made a copy past to be sure but I got the issue, here is the code


```

import toga
from toga.style import Pack
from toga.style.pack import COLUMN, ROW


class dictionnaire(toga.App):
    def startup(self):
        main_box = toga.Box(style=Pack(direction=COLUMN))

        name_label = toga.Label(
            "Your name: ",
            style=Pack(padding=(0, 5)),
        )
        self.name_input = toga.TextInput(style=Pack(flex=1))

        name_box = toga.Box(style=Pack(direction=ROW, padding=5))
        name_box.add(name_label)
        name_box.add(self.name_input)

        button = toga.Button(
            "Say Hello!",
            on_press=self.say_hello,
            style=Pack(padding=5),
        )

        main_box.add(name_box)
        main_box.add(button)

        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = main_box
        self.main_window.show()

    def say_hello(self, widget):
        print(f"Hello, {self.name_input.value}")


def main():
    return dictionnaire() 
```

**Reply by freakboy3742** [reply]
That's the code the tutorial directs you to use, but the issue isn't the code. As I indicated in my original response, It's how you're driving Briefcase. 

You've either not saved this file (so, your editor has the content, but it's not saved, so on disk the file is the old content), or you're not using the *exact* command lines that the Briefcase tutorial is telling you to use - most notably, the `-u` option to the `run` command, or the `update` command. 

However, at this point, you haven't even told us where you are in the tutorial, so I don't know if you're having issues with `briefcase run` or `briefcase dev`.

**Comment by ochampi** [comment]
ok, yes your right, the save was not up. thanks. as usual I feel stupid but thanks you
